var/spack/repos/builtin/packages/py-networkit/package.py

```diff
@@ -39,7 +41,7 @@ class PyNetworkit(PythonPackage):
     depends_on('ninja', type='build')
     depends_on('py-cython', type='build')
     depends_on('py-numpy', type=('build', 'run'))
-    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-scipy@:1.6.3', type=('build', 'run'))
```

Comment:
Does it not work with the latest scipy? Has this been reported to the developers?

Reply:
Not yet reported, I was kind of in a hurry last week. Will do so. 

As a copy: On my testing machine (macOS 10.14.6, x86_64) py-scipy 1.7.0 didn't work at all. If I install with `spack install -v py-scipy` version 1.7.0 complains about a missing `libgomp` (which is present):

```
Running from SciPy source directory.
Traceback (most recent call last):
  File "setup.py", line 629, in <module>
    setup_package()
  File "setup.py", line 625, in setup_package
    setup(**metadata)
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/core.py", line 135, in setup
    config = configuration()
  File "setup.py", line 536, in configuration
    config.add_subpackage('scipy')
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/misc_util.py", line 1016, in add_subpackage
    config_list = self.get_subpackage(subpackage_name, subpackage_path,
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/misc_util.py", line 982, in get_subpackage
    config = self._get_configuration_from_setup_py(
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/misc_util.py", line 924, in _get_configuration_from_setup_py
    config = setup_module.configuration(*args)
  File "scipy/setup.py", line 12, in configuration
    config.add_subpackage('interpolate')
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/misc_util.py", line 1016, in add_subpackage
    config_list = self.get_subpackage(subpackage_name, subpackage_path,
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/misc_util.py", line 982, in get_subpackage
    config = self._get_configuration_from_setup_py(
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-numpy-1.21.1-abjv6fh5httb24a565jradcria3apq2h/lib/python3.8/site-packages/numpy/distutils/misc_util.py", line 924, in _get_configuration_from_setup_py
    config = setup_module.configuration(*args)
  File "scipy/interpolate/setup.py", line 58, in configuration
    from pythran.dist import PythranExtension
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/__init__.py", line 41, in <module>
    from pythran.toolchain import (generate_cxx, compile_cxxfile, compile_cxxcode,
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/toolchain.py", line 6, in <module>
    from pythran.backend import Cxx, Python
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/backend.py", line 7, in <module>
    from pythran.analyses import LocalNodeDeclarations, GlobalDeclarations, Scope
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/analyses/__init__.py", line 12, in <module>
    from .aliases import Aliases, StrictAliases
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/analyses/aliases.py", line 6, in <module>
    from pythran.syntax import PythranSyntaxError
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/syntax.py", line 7, in <module>
    from pythran.tables import MODULES
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/pythran/tables.py", line 4524, in <module>
    omp_version = getattr(__import__('omp'), 'VERSION', 45)
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/omp/__init__.py", line 122, in <module>
    sys.modules[__name__] = OpenMP()
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/omp/__init__.py", line 33, in __init__
    self.init_not_msvc()
  File "/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/py-pythran-0.9.12-nukzviabg5blgfmdnsfs3rnhmmtu664l/lib/python3.8/site-packages/omp/__init__.py", line 102, in init_not_msvc
    raise ImportError("I can't find a shared library for libgomp,"
ImportError: I can't find a shared library for libgomp, you may need to install it or adjust the LD_LIBRARY_PATH environment variable.
==> Error: ProcessError: Command exited with status 1:
    '/Users/fbt/Documents/maks/networkit/repos/spack_repos/fork/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/python-3.8.11-cdpe36vo7jz4acjtk5dng3vsb6vk7s4p/bin/python3.8' '-s' 'setup.py' '--no-user-cfg' 'build'
See build log for details:
  /var/folders/rl/rhhzmk2x7pn3vq06tl745_lm0000gq/T/fbt/spack-stage/spack-stage-py-scipy-1.7.0-inmtjydjjnmktbx3p3cmoyms6lgv4ke6/spack-build-out.txt
```

Reply:
Let me see if I can debug the failing scipy installation. If it's an issue with scipy and not with networkit I would rather correct it in the scipy package.

Reply:
I'm thinking this is an issue only for `%apple-clang` which doesn't come with OpenMP libraries. I'll probably use a similar hack as I did for `py-scikit-learn` where I add a dependency on `llvm-openmp`. Will test on Wed or Thu after I get back to my Macbook.

Reply:
Thanks for investigating.

Since you are listed as maintainer for the `py-scipy` package, should I also create an issue?

Reply:
Yes please, just ping me on there.

Reply:
This _should_ be fixed by #25137, although I'm encountering CI issues that suggest that it isn't completely fixed. If you remove this change from the PR I can merge.